packages/@aws-cdk/aws-apigateway/lib/stage.ts

```diff
@@ -273,6 +273,26 @@ export class Stage extends Resource implements IStage {
     return `https://${this.restApi.restApiId}.execute-api.${Stack.of(this).region}.${Stack.of(this).urlSuffix}/${this.stageName}${path}`;
   }
 
+  /**
+   * Returns the resource ARN for this stage:
+   *
+   *   arn:aws:apigateway:{region}::/restapis/{restApiId}/stages/{stageName}
+   *
+   * Note that this is separate from the execute-api ARN for methods and resources
+   * within this stage.
+   *
+   * @attribute
+   */
+  public get stageArn() {
+    return Stack.of(this).formatArn({
+      arnFormat: ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+      service: 'apigateway',
+      account: '',
+      resource: 'restapis',
+      resourceName: `${this.restApi.restApiId}/stages/${this.stageName}`,
```

Comment:
What if `this.stageName` is unresolved? Can that happen?

Reply:
It _should_ always be a `Ref` to the `CfnStage` ([L257](https://github.com/kylelaker/aws-cdk/blob/add-apigateway-arns/packages/%40aws-cdk/aws-apigateway/lib/stage.ts#L257)), which would be safe to concatenate here. I'm not sure if it'd be possible for it to be any other type of token since it's not based on the input via `StageOptions`/`StageProps` directly.